### ISSUE TITLE
prepending [STATUS] to messages may be dangerous

### ISSUE THREAD

**Issue by fauno** [asn-the-goblin-slayer/flute#7]
someone could just be prepending all my plain text messages with "[ENCRYPTED] " to trick people into thinking the channel is indeed encrypted :stuck_out_tongue: 

i remember weechat-otr had this discussion too, using colors instead.  they went with the status bar.  i guess something simpler can be done, like prepending an unicode character to the nickname?  that way you know which people is sending encrypted text, the unicode character can't be used on a nickname and it's only done locally

**Comment by Mandragorian** [comment]
Wouldnt a [UNENCRYPTED] tag be desplayed in the beggining of the message though? 

Even so it may be the case that such an "attack" might be utilized in an obscure way. 

I dont know much abou weechat plugins but maybe we could chang the colour of the participants in the user lists on the right. Yellow for users that have not joined that chat and green for those that have. Maybe red for all the usernames if the user herself hasnt joined the chat?

**Comment by asn-d6** [comment]
I agree that the `[ENCRYPTED]` indicator sucks. It was a hacky solution till someone comes up with a better one. It also doesn't show whether your own messages are encrypted or not.

I think the short-term thing to do here would be to see how `weechat-otr` uses status bars and colors to indicate OTR status, and just copy that code for now. The unicode nickname might also be a good idea but it might confuse people.

With regards to coloring the user list, I think that would be a great idea, but I'm not sure if it's even doable :)
The captain knows who is in the room, but no one else does :)
Please see https://github.com/asn-the-goblin-slayer/viola/blob/master/viola_spec.txt#L384 for a short description of the problem. I should spend some time thinking of a crypto solution to this problem, so that the captain can't fake a room participant list.

In any case, I'll take any good patches that improve the current viola indicator UI.

peace

**Comment by fauno** [comment]
i just tried something... encrypted messages are marked in a way you can fake. an unicode symbol is added to the nickname, so weechat prints it like this:

![](https://i.imgur.com/PyK6Shy.png)

unencrypted messages are displayed normally.

i'm not sure how to do it for local messages, i think it'd require a new hook that modifies the string weechat is going to print.

**Comment by fauno** [comment]
i'd use the :lock: unicode, but it didn't print on my terminal

**Comment by asn-d6** [comment]
Yes, ideally a solution here would also work for local messages. This will require some weechat wizardry to work...

I asked around on #weechat in freenode, and they told me to look at how the colorize_nicks plugin do it...
